r1cs/gadgets/std/src/groups/group_vec.rs

```diff
@@ -0,0 +1,402 @@
+            res_vec.extend_from_slice(el_to_bytes.as_slice());
+        }
+        Ok(res_vec)
+    }
+}
+
+impl<ConstraintF: Field, G: Group, GG: GroupGadget<G, ConstraintF>> EqGadget<ConstraintF> for GroupGadgetVec<ConstraintF, G, GG> {
+    fn is_eq<CS: ConstraintSystemAbstract<ConstraintF>>(&self, mut cs: CS, other: &Self) -> Result<Boolean, SynthesisError> {
+        if self.len() != other.len() {
+            Err(SynthesisError::Unsatisfiable)?
+        }
+        let mut are_elements_eq = Vec::with_capacity(self.len());
+        for (i, (el1, el2)) in self.iter().zip(other.iter()).enumerate() {
+            let is_eq = el1.is_eq(cs.ns(|| format!("self[{}] == other[{}]", i,i)), el2)?;
+            are_elements_eq.push(is_eq);
+        }
+        Boolean::kary_and(cs.ns(|| "compute equality flag from element wise flags"), &are_elements_eq)
+    }
+}
+
+impl<ConstraintF: Field, G: Group, GG: GroupGadget<G, ConstraintF>> ToBitsGadget<ConstraintF> for GroupGadgetVec<ConstraintF, G, GG> {
+    fn to_bits<CS: ConstraintSystemAbstract<ConstraintF>>(&self, mut cs: CS) -> Result<Vec<Boolean>, SynthesisError> {
+        let mut res_vec = Vec::new();
+        for (i, el) in self.iter().enumerate() {
+            let el_to_bits = el.to_bits(cs.ns(|| format!("convert element {} to bits", i)))?;
+            res_vec.extend_from_slice(el_to_bits.as_slice());
+        }
+        Ok(res_vec)
+    }
+
+    fn to_bits_strict<CS: ConstraintSystemAbstract<ConstraintF>>(&self, mut cs: CS) -> Result<Vec<Boolean>, SynthesisError> {
+        let mut res_vec = Vec::new();
+        for (i, el) in self.iter().enumerate() {
+            let el_to_bits = el.to_bits_strict(cs.ns(|| format!("convert element {} to bits with check", i)))?;
+            res_vec.extend_from_slice(el_to_bits.as_slice());
+        }
+        Ok(res_vec)
+    }
+}
+
+impl<ConstraintF: Field, G: Group, GG: GroupGadget<G, ConstraintF>> CondSelectGadget<ConstraintF> for GroupGadgetVec<ConstraintF, G, GG> {
+    fn conditionally_select<CS: ConstraintSystemAbstract<ConstraintF>>(mut cs: CS, cond: &Boolean, first: &Self, second: &Self) -> Result<Self, SynthesisError> {
+        let mut res_vec = Vec::new();
+        for (i,(el1, el2)) in first.iter().zip(second.iter()).enumerate() {
+            res_vec.push(GG::conditionally_select(cs.ns(|| format!("cond select element {}", i)), cond, el1, el2)?);
+        }
+        let zero = GG::zero(cs.ns(|| "alloc zero"))?;
+        if first.len() > second.len() {
+            for (i, el) in first.iter().enumerate().skip(second.len()) {
+                res_vec.push(GG::conditionally_select(cs.ns(|| format!("cond select element {}", i)), cond, el, &zero)?)
+            }
+        } else {
+            for (i, el) in second.iter().enumerate().skip(first.len()) {
+                res_vec.push(GG::conditionally_select(cs.ns(|| format!("cond select element {}", i)), cond, &zero, &el)?)
+            }
+        };
```

Comment:
Since I change the zero representation to be the empty vector, now it is possible to let the conditionally select return an error if the two input vectors have different length, except for the case when at least one of them is the empty vector, which needs to be explicitly handled. Therefore, I removed this portion of the code and I added another piece which deals with either of the two vectors being the empty one before checking that the two vectors have the same length